Sources/General/KF.swift

```diff
@@ -0,0 +1,982 @@
+    @discardableResult
+    public func set(to imageView: KFCrossPlatformImageView) -> DownloadTask? {
+        imageView.kf.setImage(
+            with: source,
+            placeholder: placeholder,
+            parsedOptions: options,
+            progressBlock: progressBlock,
+            completionHandler: resultHandler
+        )
+    }
+
+    /// Builds the image task request and sets it to an `NSTextAttachment` object.
+    /// - Parameters:
+    ///   - attachment: The text attachment object which loads the task and should be set with the image.
+    ///   - attributedView: The owner of the attributed string which this `NSTextAttachment` is added.
+    /// - Returns: A task represents the image downloading, if initialized.
+    @discardableResult
+    public func set(to attachment: NSTextAttachment, attributedView: KFCrossPlatformView) -> DownloadTask? {
+        let placeholderImage = placeholder as? KFCrossPlatformImage ?? nil
+        return attachment.kf.setImage(
+            with: source,
+            attributedView: attributedView,
+            placeholder: placeholderImage,
+            parsedOptions: options,
+            progressBlock: progressBlock,
+            completionHandler: resultHandler
+        )
+    }
+
+    #if canImport(UIKit)
+
+    /// Builds the image task request and sets it to a button.
+    /// - Parameters:
+    ///   - button: The button which loads the task and should be set with the image.
+    ///   - state: The button state to which the image should be set.
+    /// - Returns: A task represents the image downloading, if initialized.
+    @discardableResult
+    public func set(to button: UIButton, for state: UIControl.State) -> DownloadTask? {
+        let placeholderImage = placeholder as? KFCrossPlatformImage ?? nil
+        return button.kf.setImage(
+            with: source,
+            for: state,
+            placeholder: placeholderImage,
+            parsedOptions: options,
+            progressBlock: progressBlock,
+            completionHandler: resultHandler
+        )
+    }
+
+    /// Builds the image task request and sets it to the background image for a button.
+    /// - Parameters:
+    ///   - button: The button which loads the task and should be set with the image.
+    ///   - state: The button state to which the image should be set.
+    /// - Returns: A task represents the image downloading, if initialized.
+    @discardableResult
+    public func setBackground(to button: UIButton, for state: UIControl.State) -> DownloadTask? {
+        let placeholderImage = placeholder as? KFCrossPlatformImage ?? nil
+        return button.kf.setBackgroundImage(
+            with: source,
+            for: state,
+            placeholder: placeholderImage,
+            parsedOptions: options,
+            progressBlock: progressBlock,
+            completionHandler: resultHandler
+        )
+    }
+    #endif // end of canImport(UIKit)
+
+    #if canImport(AppKit)
```

Comment:
I had to change this to `#if canImport(AppKit) && !targetEnvironment(macCatalyst)`